### PR TITLE
A number of tiny enhancements

### PR DESCRIPTION
**Pull request by famz**
Stefan, I'm migrating from my "git-publish" script to yours, these are the results of two weeks' using and tweaking. Could you take a look?

### CODE REVIEW

**Comment by stefanha** [comment]
How do you use --in-reply-to?  It seems like it would resend the entire series with the In-Reply-To header, which isn't how [PATCH 11/10] emails work.

**Comment by stefanha** [comment]
The publish-cc-list hook seems similar to the git-send-email(1) --cc-cmd option.  Please implement that instead for consistency.

**Comment by stefanha** [comment]
Please drop the environment variable for the pre-publish-tag hook.  githooks(5) take positional arguments.  None of the other git-publish hooks take environment variables either.

Please stay consistent, especially since the difference between BASE="$1" inside the script vs passing the BASE environment variable is trivial.

**Comment by famz** [comment]
I'll address these comments and update the PR once the I'm happy with it, maybe after several days.

Regarding how to use --in-reply-to:

    # Usual publish
    git -p qemu -b master
    [Sent 10 patches here]
    # After committed a supplementary patch
    git publish -b HEAD~1 -p qemu --in-reply-to=$message_id_of_previous_cover_letter --num 1 --prefix "PATCH 11/10"

**Comment by famz** [comment]
I don't know how to do the same functionality with git-send-email --cc-cmd. Ideally, the exhaustive list of Cc'ed addressed should appear at --inspect-emails point, and with the "Add inspect email operations for editing Cc list" feature, it should be editable.

Whereas with git-send-email --cc-cmd, it is too late.

**Comment by stefanha** [comment]
On Fri, Jul 22, 2016 at 9:29 AM, Fam Zheng <notifications@github.com> wrote:
>
> I don't know how to do the same functionality with git-send-email --cc-cmd. Ideally, the exhaustive list of Cc'ed addressed should appear at --inspect-emails point, and with the "Add inspect email operations for editing Cc list" feature, it should be editable.
>
> Whereas with git-send-email --cc-cmd, it is too late.

Okay, then git-publish --cc-cmd should run before git-send-email is
invoked.  This makes the git-publish command-line interface similar to
git-send-email --cc-cmd.

**Comment by famz** [comment]
Stefan, I've updated the branch. The last three commits are new so it's less tested.

File: git-publish [hunk 1]
```diff
@@ -594,14 +561,37 @@ def main():
...
+            git_send_email(to, cc, tmpdir, suppress_cc, options.in_reply_to)
         except (GitSendEmailError, GitHookError, InspectEmailsError):
             return 1
+        finally:
+            if tmpdir:
+                shutil.rmtree(tmpdir)
 
         # Store --to and --cc for next revision
         git_set_config('branch', topic, 'gitpublishto', to)
+        print options.override_cc
```
Comment (stefanha) [inline git-publish]: Is this debugging code that should be removed?

File: git-publish [hunk 2]
```diff
@@ -393,6 +393,8 @@ def parse_args():
                       help='branch which this is based off [defaults to master]')
     parser.add_option('--cc', dest='cc', action='append', default=[],
                       help='specify a Cc: email recipient')
+    parser.add_option('--cc-cmd',
+                      help='specify a command to to append to the cc list.')
```
Comment (stefanha) [inline git-publish]: Other help messages don't end with punctuation, please drop the  '.'

File: git-publish (hunk 1 above)
Comment (famz) [inline git-publish]: Yes.

File: git-publish (hunk 2 above)
Comment (famz) [inline git-publish]: Yes. Will fix the 'to to' as well.

**Comment by famz** [comment]
I fixed both issues. Thanks!

**Pull request merged**